src/library/scala/collection/immutable/Vector.scala

```diff
@@ -31,13 +31,47 @@ object Vector extends StrictOptimizedSeqFactory[Vector] {
 
   def empty[A]: Vector[A] = NIL
 
-  def from[E](it: collection.IterableOnce[E]): Vector[E] =
+  def applyOld[A](elems: A*): Vector[A] = fromOld[A](elems)
+
+  def fromOld[E](it: collection.IterableOnce[E]): Vector[E] =
     it match {
       case v: Vector[E] => v
       case _ if it.knownSize == 0 => empty[E]
       case _ => (newBuilder ++= it).result()
     }
 
+  def from[E](it: collection.IterableOnce[E]): Vector[E] =
+    it match {
+      case as: ArraySeq[E] if as.length <= 32 && as.unsafeArray.isInstanceOf[Array[AnyRef]] =>
+        if (as.isEmpty) NIL
+        else {
+          val v = new Vector(0, as.length, 0)
+          v.display0 = as.unsafeArray.asInstanceOf[Array[AnyRef]]
+          v.depth = 1
+          v
+        }
+      case v: Vector[E] => v
+      case _ =>
+        val knownSize = it.knownSize
+
+        if (knownSize == 0) empty[E]
+        else if (knownSize > 0 && knownSize <= 32) {
+          var i = 0
```

Comment:
How does this impl perform compared to `it.iterator.toArray`?

Reply:
Will definitely try it out!

Reply:
I couldn't really do it so far as I know because I need to cast each element from `E` to `AnyRef`, unless there's an other workaround. (aka toArray returns Array[E] not Array[AnyRef] so compile error)